test/functional/wallet_listdescriptors.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the listdescriptors RPC."""
+
+from test_framework.descriptors import (
+    descsum_create
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+
+class ReceivedByTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    # do not create any wallet by default
+    def init_wallet(self, i):
+        return
+
+    def run_test(self):
+        node = self.nodes[0]
+        assert_raises_rpc_error(-18, 'No wallet is loaded.', node.listdescriptors)
+
+        self.log.info('Test that the command is not available for legacy wallets.')
+        node.createwallet(wallet_name='w1', descriptors=False)
+        assert_raises_rpc_error(-4, 'listdescriptors is not available for non-descriptor wallets', node.listdescriptors)
+
+        self.log.info('Test the command for empty descriptors wallet.')
+        node.createwallet(wallet_name='w2', blank=True)
+        assert_equal(0, len(node.get_wallet_rpc('w2').listdescriptors()))
+
+        self.log.info('Test the command for a default descriptors wallet.')
+        node.createwallet(wallet_name='w3')
+        result = node.get_wallet_rpc('w3').listdescriptors()
+        assert_equal(6, len(result))
+        assert_equal(6, len([d for d in result if d['active']]))
+        assert_equal(3, len([d for d in result if d['internal']]))
+        for item in result:
+            assert item['desc'] is not ''
```

Comment:
linter isn't liking this:
`test/functional/wallet_listdescriptors.py:47:20: F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)`

Reply:
fixed